bash-preexec.sh

```diff
@@ -256,7 +296,7 @@ __bp_preexec_invoke_exec() {
         # Only execute each function if it actually exists.
         # Test existence of function with: declare -[fF]
         if type -t "$preexec_function" 1>/dev/null; then
-            __bp_set_ret_value ${__bp_last_ret_value:-}
+            __bp_set_ret_value "$lastexit" "$lastarg"
```

Comment:
Is there a reason to add the lastarg value here? This wasn't being set prior no?

Reply:
Oh, OK. Right. The last argument `$_` has not been set for `preexec` before.

I have added it for consistency with `precmd_functions`. It seems to be introduced for `precmd` in d056f7dc37e30724f09eb7487a060311caf49990 via PR #32. Is there a reason that we do not provide `preexec` functions with `$_` while  we provide `precmd` functions with `$_`? As a side note, similar functionalities in ble.sh, blehook PRECMD/PREEXEC/POSTEXEC, all set `$_` for the hook functions.

In PR #32, Zsh's behavior is referenced for `precmd`. I have now checked Zsh's behavior for `preexec`. Zsh seems to set `$_` also for `preexec`:

```zsh
$ zsh
% preexec1() { echo "preexec1: lastarg='$_'"; }
% preexec2() { echo "preexec2: lastarg='$_'"; }
% add-zsh-hook preexec preexec1
% add-zsh-hook preexec preexec2
preexec1: lastarg='preexec1'
% echo hello
preexec1: lastarg='preexec2'
preexec2: lastarg='preexec2'
hello
% echo world
preexec1: lastarg='hello'
preexec2: lastarg='hello'
world
```